core/demangle_ocaml_symbols.ml

```diff
@@ -0,0 +1,50 @@
+open! Core
+
+let decode_two_digit_hexadecimal_number first_character second_character =
+  let%bind.Option hex_digit_of_first_character = Char.get_hex_digit first_character in
+  let%bind.Option hex_digit_of_second_character = Char.get_hex_digit second_character in
+  let leftshift_first_character = Int.shift_left hex_digit_of_first_character 4 in
+  let bit_or_on_the_hexadecimals =
+    Int.bit_or leftshift_first_character hex_digit_of_second_character
+  in
+  match Char.of_int bit_or_on_the_hexadecimals with
```

Comment:
This seems equivalent to not having the `match` at all, unless I'm misreading this.

Reply:
Thank you for the suggestion! Realized the match statement was not necessary as it wasn't doing anything and removing did the same thing.  I fixed all the other suggestions too.